purchases/src/main/kotlin/com/revenuecat/purchases/Offerings.kt

```diff
@@ -43,31 +44,48 @@ data class Offerings internal constructor(
         val showNoOffering = placements.offeringIdsByPlacement.containsKey(placementId)
         val offering = offeringForPlacement ?: if (showNoOffering) null else fallbackOffering
 
-        return offering?.withPlacement(placementId)
+        return offering?.withPresentedContext(placementId = placementId, targeting = this.targeting)
     }
-}
 
-/**
- * This class contains all the offerings by placement configured in RevenueCat dashboard.
- * For more info see https://www.revenuecat.com/docs/targeting
- * @property fallbackOfferingId The optional offering identifier to fallback on if the placement isn't found.
- * @property offeringIdsByPlacement Dictionary of all offering identifiers keyed by their placement identifier.
- */
-internal data class Placements(
-    val fallbackOfferingId: String?,
-    val offeringIdsByPlacement: Map<String, String?>,
-)
+    /**
+     * This class contains information about the targeting that presented the offering.
+     * @property revision The revision of the targeting that presented the offering.
+     * @property ruleId The rule id of the targeting that presented the offering.
+     */
+    internal data class Targeting(
+        val revision: Int,
+        val ruleId: String,
+    )
 
-private fun Offering.withPlacement(placementId: String): Offering {
+    /**
+     * This class contains all the offerings by placement configured in RevenueCat dashboard.
+     * For more info see https://www.revenuecat.com/docs/targeting
+     * @property fallbackOfferingId The optional offering identifier to fallback on if the placement isn't found.
+     * @property offeringIdsByPlacement Dictionary of all offering identifiers keyed by their placement identifier.
+     */
+    internal data class Placements(
+        val fallbackOfferingId: String?,
+        val offeringIdsByPlacement: Map<String, String?>,
+    )
+}
+
+internal fun Offering.withPresentedContext(placementId: String?, targeting: Offerings.Targeting?): Offering {
     val updatedAvailablePackages = this.availablePackages.map {
-        val context = it.presentedOfferingContext.copy(placementIdentifier = placementId)
-        val product = it.product.copyWithPresentedOfferingContext(context)
+        val oldContext = it.presentedOfferingContext
+
+        val newContext = oldContext.copy(
+            placementIdentifier = placementId ?: oldContext.placementIdentifier,
```

Comment:
This means this method can't set the value to null, even if given as a parameter... My other suggestion was to pass the original value of the offering as default values of the parameters of this method. Then, if null is passed, we know it's explicit, and we set it here. NABD though, we probably don't really need to set it to null if it has a value I think?